### PR TITLE
Feature fix ci

### PR DESCRIPTION
**Pull request by VoR0220**
The following fixes the CI to actually RUN tests. During the consolidation this was broken due to what we now consider unnecessary multiple docker machine runs. Since we are moving to deprecate docker machine, this presents a new CI for appveyor and travis that will:

1. Mandate passing of one docker machine instance
2. Run but allow the failure of a native docker solution (docker for windows, docker for mac) so that we can work towards these native solutions while they are still being developed on for better support. 

Also included in this is a separation of docker-machine script from the normal test.sh script...all tests can be run now from the test.sh script with minimal effort applied. 

Currently the tests do not pass, but this is not the point of this PR. The point of this PR is to get the CI working again and running properly. Failing tests will be dealt with in future PRs right after this one is merged.

### CODE REVIEW

File: tests/README.md [hunk 1]
```diff
@@ -16,7 +17,7 @@ This is a hard tool to test. In order to clearly define (and limit) the testing
 
 ## Goal 1: Test The Tool Itself
 
-Testing the tool itself is performed from the `tests/test_tool.sh` script. This script should only concern itself with the mechanisms for testing the tool. This script should be run typically from inside the eris/eris container (but I also use it for quick testing of the package level tests locally — see below).
+Testing the tool itself is performed from the `tests/test_tool.sh` script. This script should only concern itself with the mechanisms for testing the tool. This script should be run typically from inside the eris/eris container unless testing on a specific os environment. It is also acceptable to use it for quick testing of the package level tests locally — see below.
 
 ## Goal 2: Test The Minimum Viable Stack Connection and Sequencing (Apps and Contract Suites)
```
Comment (zramsay) [inline tests/README.md]: this whole file can go, it'll be replaced by the file i'm writing for the ops repo. it's also super outdated

File: tests/machines/docker.sh [hunk 2]
```diff
@@ -8,9 +8,6 @@
 # we need to be on a specific version of docker. It will likely be unuseful to
 # you.
 
-# If you are looking for a quick and easy way to set up eris on a cloud machin
```
Comment (zramsay) [inline tests/machines/docker.sh]: ++

File: tests/machines/docker_machine.sh [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+
+# remove the machines
+function remove_machine() {
+  if [ "$ci" = true ]
+  then
+    docker-machine rm --force $(docker-machine ls -q)
+  else
+    docker-machine rm --force $machine
+  fi
+}
```
Comment (zramsay) [inline tests/machines/docker_machine.sh]: much cleaner than the setup.go

File: tests/test.sh [hunk 4]
```diff
@@ -52,208 +47,26 @@ BRANCH=${CIRCLE_BRANCH:=master}
...
-
-# ----------------------------------------------------------------------------
-# Build eris in a docker images
-
-# build docker images; runs as background process
-build_eris() {
-  echo "Building eris in a docker container."
-  cd $repo
-  export BRANCH=$1
-  tests/build_tool.sh &>/dev/null
```
Comment (zramsay) [inline tests/test.sh]: has this script been removed? if not it should

File: tests/test.sh [hunk 5]
```diff
@@ -52,208 +47,26 @@ BRANCH=${CIRCLE_BRANCH:=master}
...
-# non current docker versions. current versions docker versions do not utilize this DinD technique
-# but rather test in the context of the "host" outer environment against a set backend.
-test_tool_in_docker() {
-  echo "Starting Eris Docker container."
-  if [ "$ci" = true ]
-  then
-    docker run --name test_tool --volume $HOME/$dm_path:/home/$testuser/$dm_path --entrypoint $entrypoint -e MACHINE_NAME=$machine -p $hostsocket:$remotesocket --user $testuser $testimage:$1 &> $CIRCLE_ARTIFACTS/$1.log
-  else
-    docker run --name test_tool --rm --volume $HOME/$dm_path:/home/$testuser/$dm_path --entrypoint $entrypoint -e MACHINE_NAME=$machine -p $hostsocket:$remotesocket --user $testuser $testimage:$1 &> $CIRCLE_ARTIFACTS/$1.log
-  fi
```
Comment (zramsay) [inline tests/test.sh]: thank f*ck this ugliness is gone

File: tests/test_stack.sh [hunk 6]
```diff
@@ -60,15 +60,15 @@ check_and_exit() {
 # ----------------------------------------------------------------------------
 # Run ECM tests
```
Comment (zramsay) [inline tests/test_stack.sh]: `run [eris chains make] tests`

File: tests/test_stack.sh [hunk 7]
```diff
@@ -60,15 +60,15 @@ check_and_exit() {
...
 
-tests/test_chains_make.sh
+time tests/test_chains_make.sh
 test_exit=$?
 check_and_exit
 cd $start
 
 # ----------------------------------------------------------------------------
 # Run EPM tests
```
Comment (zramsay) [inline tests/test_stack.sh]: `run [eris pkgs do] tests`

File: tests/test_tool.sh [hunk 8]
```diff
@@ -111,29 +114,29 @@ setup() {
...
+    go test ./chains/... -v -timeout 20m && passed Chains
     if [ $? -ne 0 ]; then return 1; fi
-    go test ./keys/... && passed Keys
+    go test ./keys/... -v -timeout 20m && passed Keys
     if [ $? -ne 0 ]; then return 1; fi
-    go test ./pkgs/... && passed Packages
+    go test ./pkgs/... -v -timeout 20m && passed Packages
     if [ $? -ne 0 ]; then return 1; fi
-    go test ./clean/... && passed Clean
+    go test ./clean/... -v -timeout 20m && passed Clean
```
Comment (zramsay) [inline tests/test_tool.sh]: we should not have to increase the timeout unless absolutely necessary (and i don't think it it...)

File: tests/README.md (hunk 1 above)
Comment (VoR0220) [inline tests/README.md]: ++

File: tests/test.sh (hunk 4 above)
Comment (VoR0220) [inline tests/test.sh]: wasn't positive if we were still going to build CLI as a docker image.

File: tests/test_tool.sh (hunk 8 above)
Comment (VoR0220) [inline tests/test_tool.sh]: It is for Windows...for whatever reason they run slow as fuck there...in particular the perform package and the data package tests.

Comment (zramsay) [inline tests/test_tool.sh]: then only add timeouts for those packages & make a note to remove

File: tests/build_other_images.sh [hunk 9]
```diff
@@ -1,48 +0,0 @@
...
-cp $repo/docker/x86/data/Dockerfile .
-
-if [[ "$branch" = "master" || "$branch" = "develop" ]]
-then
-  docker build -t $testimage:latest .
-  docker tag $testimage:latest $testimage:$release_maj
-  docker tag $testimage:latest $testimage:$release_min
-else
-  docker build -t $testimage:$release_min .
-fi
```
Comment (zramsay) [inline tests/build_other_images.sh]: whoa whoa, this is needed & also represents a missing piece of the CI pipeline, namely, the automation of the build + push of a versioned `quay.io/eris/data` image. Should be handled by Jenkins; the building/testing/pushing flow is still being fleshed out.

File: circle.yml [hunk 10]
```diff
@@ -1,43 +0,0 @@
...
-    branch: master
-    commands:
-      - docker rmi quay.io/eris/eris:master
-      - docker login -e $DOCKER_EMAIL -u $DOCKER_USER -p $DOCKER_PASS quay.io
-      - docker push quay.io/eris/eris
-      - docs/build.sh master
-  develop:
-    branch: develop
-    commands:
-      - docs/build.sh
```
Comment (zramsay) [inline circle.yml]: what's gonna run the docs/build.sh script now?

File: tests/build_other_images.sh (hunk 9 above)
Comment (VoR0220) [inline tests/build_other_images.sh]: https://www.youtube.com/watch?v=2MpuwUHeNFc

File: circle.yml (hunk 10 above)
Comment (VoR0220) [inline circle.yml]: Jenkins?

File: tests/test.sh [hunk 11]
```diff
@@ -52,208 +47,26 @@ BRANCH=${CIRCLE_BRANCH:=master}
 BRANCH=${BRANCH/-/_}
 BRANCH=${BRANCH/\//_}
 
-declare -a MACHINES
-declare -a MACH_RESULTS=()
-
 # Define now the tool tests within the Docker container will be booted from docker run
-entrypoint="/home/eris/test_tool.sh"
+entrypoint="$GOPATH/src/github.com/eris-ltd/eris/tests/test_tool.sh"
 testimage=quay.io/eris/eris
```
Comment (zramsay) [inline tests/test.sh]: aren't we getting rid of this?

Comment (VoR0220) [inline tests/test.sh]: Good point. Missed that.

**Pull request merged**